assets/js/googlesitekit/notifications/datastore/notifications.js

```diff
@@ -142,6 +164,74 @@ export const actions = {
 			},
 			type: REGISTER_NOTIFICATION,
 		};
+
+		const registry = yield commonActions.getRegistry();
+
+		// If no view contexts were provided, we should instead do a comparison
+		// with the "top"/"visible" notification in the queue for this
+		// notification's `groupID` to see if the newly-registered notification
+		// should be added to the "top"/"visible position" in the queue.
+		//
+		// This is the usual route for "ad-hoc" notifications that are registered
+		// after initial page load, such as Setup Success notifications.
+		if ( ! viewContexts?.length ) {
+			// To do this, we'll add the notification to the queue immediately, which
+			// will insert it in the right position based on its priority and other
+			// criteria (handled by the reducer for this action).
+			registry
+				.dispatch( CORE_NOTIFICATIONS )
+				.addNotificationToQueue( id );
```

Comment:
This should always use `yield actions.addNotificationToQueue( id )` instead for actions on the same store. Or this action could be independent of `actions` if we don't want to expose it on the registry. We shouldn't need `registry` I don't think.